### ISSUE TITLE
New installed theme does not persist

### ISSUE THREAD

**Issue by RickStanley** [LunarVim/LunarVim#2408]
### Problem description

Steps to reproduce behaviour:
1. open LunarVim;
2. open "Configuration";
3. add `"sainnhe/gruvbox-material"` to plugins;
4. save and wait packer sync;
5. change `lvim.colorscheme` from "onedarker" to "gruvbox-material";
6. save;
7. close LunarVim;
8. reopen;
9. see original theme (onedarker).

### LunarVim version

master-23feb96

### Neovim version (>= 0.6.1)

NVIM v0.6.1 Build type: Release LuaJIT 2.1.0-beta3

### Operating system/version

Linux 5.17.1-arch1-1 #1 SMP PREEMPT Mon, 28 Mar 2022 20:55:33 +0000 x86_64 GNU/Linux

### Relevant log output

```shell
null-ls: require("null-ls.health").check()
========================================================================
  - INFO: no sources registered

nvim: health#nvim#check
========================================================================
## Configuration
  - WARNING: Missing user config file: /home/stanley/.config/nvim/init.vim
    - ADVICE:
      - :help |init.vim|

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $VTE_VERSION='6800'
  - INFO: $COLORTERM='truecolor'

nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - OK: neovim version >= 0.6.0
  - WARNING: **Go**: not available
  - WARNING: **Composer**: not available
  - WARNING: **julia**: not available
  - OK: **bash**: `GNU bash, Version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **gzip**: `gzip 1.11`
  - OK: **curl**: `curl 7.82.0 (x86_64-pc-linux-gnu) libcurl/7.82.0 OpenSSL/1.1.1n zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.47.0`
  - OK: **wget**: `GNU Wget 1.21.3 übersetzt unter linux-gnu.`
  - OK: **python3**: `Python 3.10.4`
  - OK: **node**: `v17.3.1`
  - OK: **PHP**: `PHP 8.1.4 (cli) (built: Mar 16 2022 11:32:47) (NTS)`
  - OK: **Ruby**: `ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]`
  - OK: **sh**: `Ok`
  - OK: **java**: `Ok`
  - OK: **RubyGem**: `3.3.8`
  - OK: **javac**: `Ok`
  - OK: **pip3**: `pip 21.2.4 from /home/stanley/.local/lib/python3.10/site-packages/pip (python 3.10)`
  - OK: **npm**: `8.3.0`

nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.1 (062421dece3315bd6f228ad6d468cba083d0a2d5) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v17.3.1 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.2.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - tsx            ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - java           ✓ ✓ . ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - javascript     ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

provider: health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: wl-copy

## Python 2 provider (optional)
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - ERROR: Python provider error:
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          python2 not found in search path or not executable.
          python2.7 not found in search path or not executable.
          python2.6 not found in search path or not executable.
          /usr/bin/python is Python 3.10 and cannot provide Python 2.
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Executable: /usr/bin/python3
  - INFO: Python version: 3.10.4
  - INFO: pynvim version: 0.4.3
  - OK: Latest pynvim is installed.

## Python virtualenv
  - OK: no $VIRTUAL_ENV

## Ruby provider (optional)
  - INFO: Ruby: ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v17.3.1
  - INFO: Nvim node.js host: /home/stanley/.nvm/versions/node/v17.3.1/lib/node_modules/neovim/bin/cli.js
  - OK: Latest "neovim" npm/yarn package is installed: 4.10.1

## Perl provider (optional)
  - ERROR: perl provider error:
    - ADVICE:
      - "Neovim::Ext" cpan module is not installed

telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.3.1

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided

## Telescope Extension: `projects`
  - INFO: No healthcheck provided

vim.lsp: require("vim.lsp.health").check()
========================================================================
  - INFO: LSP log level : WARN
  - INFO: Log path: /home/stanley/.cache/nvim/lsp.log
  - INFO: Log size: 1 KB

vim.treesitter: require("vim.treesitter.health").check()
========================================================================
  - INFO: Runtime ABI version : 14
  - OK: Loaded parser for bash: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for java: ABI version 13
  - OK: Loaded parser for javascript: ABI version 13
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for python: ABI version 13
  - OK: Loaded parser for rust: ABI version 13
  - OK: Loaded parser for tsx: ABI version 13
  - OK: Loaded parser for typescript: ABI version 13
  - OK: Loaded parser for yaml: ABI version 13

which_key: health#which_key#check
========================================================================
## WhichKey: checking conflicting keymaps
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>Lc"**
  - INFO: old rhs: `<CMD>edit /home/stanley/.config/lvim/config.lua <CR>`
  - INFO: new rhs: `<cmd>edit /home/stanley/.config/lvim/config.lua<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>f"**
  - INFO: old rhs: `<CMD>Telescope find_files<CR>`
  - INFO: new rhs: `<cmd>lua require("which-key").execute(1)<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>st"**
  - INFO: old rhs: `<CMD>Telescope live_grep<CR>`
  - INFO: new rhs: `<cmd>Telescope live_grep<cr>`
  - WARNING: buffer-local keymap overriding global for mode **"n"**, buf: 0, lhs: **"<leader>sr"**
  - INFO: old rhs: `<CMD>Telescope oldfiles<CR>`
  - INFO: new rhs: `<cmd>Telescope oldfiles<cr>`
```

**Comment by cpea2506** [comment]
Try `PackerSync` again to completely remove `onedarker` out of your plugins. This way the new theme will load as expected.

**Comment by abzcoding** [comment]
@kylo252 did we merge the theme fix into master?

**Comment by kylo252** [comment]
The main problem is that we're not actually removing `onedarker.nvim`, here's the explanation I wrote earlier on the discord channel about this

>  it's a bit of a hidden internal tweak, that's related to two parts:
> - packer.sync was removed from lvim.reload quite a while ago now, because it was deemed too unstable to be called like that, instead: packer.install was called to add any missing plugins
> - in a recent refactor lunarvim#2351, I changed the default behavior of requiring "core" plugins, to rely on packer.snapshot API instead
> 
> we might be able to bring back packer.sync again into lvim.reload, now that it's dealing with fewer plugins, or we could wait until that feature is implemented in packer.snapshot (which is also missing another important piece of puzzle https://github.com/wbthomason/packer.nvim/issues/862)

**Comment by kylo252** [comment]
@abzcoding, a quick fix for this is to wrap calling the setup function with a check if `lvim.colorscheme` matches
https://github.com/LunarVim/LunarVim/blob/7807f147c192fdb6aec7659a3098b4bf31852fcd/lua/lvim/plugins.lua#L13-L20

**Comment by abzcoding** [comment]
> @abzcoding, a quick fix for this is to wrap calling the setup function with a check if `lvim.colorscheme` matches
> 
> https://github.com/LunarVim/LunarVim/blob/7807f147c192fdb6aec7659a3098b4bf31852fcd/lua/lvim/plugins.lua#L13-L20

we should probably bring back `:PackerSync` or do the other thingy that you said, till the snapshot bug is fixed

**Comment by kylo252** [comment]
@RickStanley, please take a look at #2410

**Comment by miguno** [comment]
Ran into the same theme issue with a fresh (stable) install today.

**Comment by abzcoding** [comment]
> Ran into the same theme issue with a fresh (stable) install today.

this would be fixed via #2410, it would first merged into `rolling` and after a bit in master

**Comment by aniketgm** [comment]
I was wondering why Packer was removing "onedarker" when I installed a different colorscheme. So is this intentional ? I mean why can't onedarker stay and I would just enable it say using something like:
```
lvim.colorscheme.onedarker = true
```
when I want and set it to `false` when I wish to switch to another colorscheme. Thoughts ?
